OtripleS.Web.Api/Services/Foundations/Teachers/TeacherService.cs

```diff
@@ -28,16 +28,16 @@ public partial class TeacherService : ITeacherService
             this.dateTimeBroker = dateTimeBroker;
         }
 
-        public ValueTask<Teacher> CreateTeacherAsync(Teacher teacher) =>
-        TryCatch(async () =>
+        public ValueTask<Teacher> CreateTeacherAsync(
+            Teacher teacher) => TryCatch(async () =>
         {
             ValidateTeacherOnCreate(teacher);
 
             return await this.storageBroker.InsertTeacherAsync(teacher);
         });
 
-        public ValueTask<Teacher> RemoveTeacherByIdAsync(Guid teacherId) =>
-        TryCatch(async () =>
+        public ValueTask<Teacher> RemoveTeacherByIdAsync(
+            Guid teacherId) => TryCatch(async () =>
```

Comment:
I saw you did that here (https://github.com/hassanhabib/OtripleS/pull/676/files) so I though we're going in that direction?

Reply:
To be honest I am more for the first approach :)

Reply:
@mtoncic Oh I see. in this part I was moving the opening parenthesis up by one.
If the method itself is over 120 characters then you wanna break the input variable to the next line. Otherwise keep it as is.

Reply:
@hassanhabib also in that example it seems you prefer 
```csharp
  public ValueTask<AssignmentAttachment> RetrieveAssignmentAttachmentByIdAsync(
            Guid assignmentId,
            Guid attachmentId) => TryCatch(async () =>
        {
            ValidateAssignmentAttachmentIds(assignmentId, attachmentId);

            AssignmentAttachment storageAssignmentAttachment =
                await this.storageBroker.SelectAssignmentAttachmentByIdAsync(assignmentId, attachmentId);

            ValidateStorageAssignmentAttachment(storageAssignmentAttachment, assignmentId, attachmentId);

            return storageAssignmentAttachment;
        });

```

over

```
  public ValueTask<AssignmentAttachment> RetrieveAssignmentAttachmentByIdAsync(
            Guid assignmentId,
            Guid attachmentId) => 
        TryCatch(async () =>
        {
            ValidateAssignmentAttachmentIds(assignmentId, attachmentId);

            AssignmentAttachment storageAssignmentAttachment =
                await this.storageBroker.SelectAssignmentAttachmentByIdAsync(assignmentId, attachmentId);

            ValidateStorageAssignmentAttachment(storageAssignmentAttachment, assignmentId, attachmentId);

            return storageAssignmentAttachment;
        });

```

Another question:
Why we're not indenting here?
```csharp
public IQueryable<AssignmentAttachment> RetrieveAllAssignmentAttachments() =>
TryCatch(() => this.storageBroker.SelectAllAssignmentAttachments());
```
as
```csharp
public IQueryable<AssignmentAttachment> RetrieveAllAssignmentAttachments() =>
        TryCatch(() => this.storageBroker.SelectAllAssignmentAttachments());
```

Reply:
@mtoncic That's a great question.
`TryCatch` is a bit of a new pattern. it's not meant to be something *within* the method so we can indent it (even though it is) but rather a wrapper around the function. Best C# could do in this day and age is to let me do a monadic trick like this and just push it as a part of the same function.